application_test.go

```diff
@@ -65,19 +65,76 @@ func TestApplicationCPU(t *testing.T) {
 
 func TestApplicationArgs(t *testing.T) {
 	app := NewDockerApplication()
-	assert.Equal(t, 0, len(app.Args))
-	app.Arg("-p").Arg("option", "-v")
-	assert.Equal(t, 3, len(app.Args))
-	assert.Equal(t, "-p", app.Args[0])
-	assert.Equal(t, "option", app.Args[1])
-	assert.Equal(t, "-v", app.Args[2])
+	assert.Nil(t, app.Args)
+	app.AddArgs("-p").AddArgs("option", "-v")
+	assert.Equal(t, 3, len(*app.Args))
+	assert.Equal(t, "-p", (*app.Args)[0])
+	assert.Equal(t, "option", (*app.Args)[1])
+	assert.Equal(t, "-v", (*app.Args)[2])
+
+	app.EmptyArgs()
+	assert.NotNil(t, app.Args)
+	assert.Equal(t, 0, len(*app.Args))
+}
+
+func ExampleApplication_AddConstraint() {
+	app := NewDockerApplication()
+
+	// add two constraints
+	app.AddConstraint("hostname", "UNIQUE").
+		AddConstraint("rack_id", "CLUSTER", "rack-1")
+}
+
+func TestApplicationConstraints(t *testing.T) {
+	app := NewDockerApplication()
+	assert.Nil(t, app.Constraints)
+	app.AddConstraint("hostname", "UNIQUE").
+		AddConstraint("rack_id", "CLUSTER", "rack-1")
+
+	assert.Equal(t, 2, len(*app.Constraints))
+	assert.Equal(t, []string{"hostname", "UNIQUE"}, (*app.Constraints)[0])
+	assert.Equal(t, []string{"rack_id", "CLUSTER", "rack-1"}, (*app.Constraints)[1])
+
+	app.EmptyConstraints()
+	assert.NotNil(t, app.Constraints)
+	assert.Equal(t, 0, len(*app.Constraints))
 }
 
 func TestApplicationEnvs(t *testing.T) {
 	app := NewDockerApplication()
-	assert.Equal(t, 0, len(app.Env))
+	assert.Nil(t, app.Env)
 	app.AddEnv("hello", "world")
```

Comment:
You could add another environment variable to verify that multiple variables are working.